dashboard/src/components/DeploymentForm/BasicDeploymentForm/DiskSizeParam.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+import { IBasicFormParam } from "shared/types";
+import Slider from "../../../components/Slider";
+import DiskSizeParam from "./DiskSizeParam";
+
+const defaultProps = {
+  id: "disk",
+  name: "diskSize",
+  label: "Disk Size",
+  param: {
+    value: "10Gi",
+    type: "string",
+    path: "disk",
+  } as IBasicFormParam,
+  handleBasicFormParamChange: jest.fn(),
+};
+
+it("renders a disk size param with a default value", () => {
+  const wrapper = shallow(<DiskSizeParam {...defaultProps} />);
+  expect(wrapper.state("Gi")).toBe(10);
+  expect(wrapper).toMatchSnapshot();
+});
+
+it("changes the value of the param when the slider changes", () => {
+  const param = {
+    value: "10Gi",
+    type: "string",
+    path: "disk",
+  } as IBasicFormParam;
+  const handleBasicFormParamChange = jest.fn(() => {
+    param.value = "20Gi";
+    return jest.fn();
+  });
+
+  const wrapper = shallow(
+    <DiskSizeParam
+      {...defaultProps}
+      param={param}
+      handleBasicFormParamChange={handleBasicFormParamChange}
+    />,
+  );
+  expect(wrapper.state("Gi")).toBe(10);
+
+  const slider = wrapper.find(Slider);
+  (slider.prop("onChange") as (values: number[]) => void)([20]);
+
+  expect(param.value).toBe("20Gi");
+  expect(handleBasicFormParamChange.mock.calls[0]).toEqual([
+    "diskSize",
+    { value: "20Gi", type: "string", path: "disk" },
+  ]);
+});
+
+it("changes only the value of the state when the slider is being updated", () => {
```

Comment:
This confused me until I read the actual test details... but I can't see an obviously better test title. Perhaps: "updates state but does not change param value during slider update" .

I think was confused me was "changes only the value of the state", because it seemed odd that we'd be updating the component state but not updating handling the form param change. After reading below, I see it's because the component encapsulates both the slider and the text input - we want to update both state and handle the change for the input, but only trigger the change for the slider when it is "dropped", which makes sense.